File: Trigger/TrigT1/TrigT1CaloMonitoring/src/JfexMonitorAlgorithm.cxx
/*
  Copyright (C) 2002-2022 CERN for the benefit of the ATLAS collaboration
*/

#include "JfexMonitorAlgorithm.h"
#include "TMath.h"

JfexMonitorAlgorithm::JfexMonitorAlgorithm( const std::string& name, ISvcLocator* pSvcLocator )
  : AthMonitorAlgorithm(name,pSvcLocator)
{
}

StatusCode JfexMonitorAlgorithm::initialize() {

  ATH_MSG_DEBUG("JfexMonitorAlgorith::initialize");
  ATH_MSG_DEBUG("Package Name "<< m_packageName);

  ATH_MSG_DEBUG("m_jFexLRJetContainerKey"<< m_jFexLRJetContainerKey);
  ATH_MSG_DEBUG("m_jFexSRJetContainerKey"<< m_jFexSRJetContainerKey);
  ATH_MSG_DEBUG("m_jFexTauContainerKey"  << m_jFexTauContainerKey);
  ATH_MSG_DEBUG("m_jFexFwdElContainerKey"<< m_jFexFwdElContainerKey);
  ATH_MSG_DEBUG("m_jFexMETContainerKey"  << m_jFexMETContainerKey);
  ATH_MSG_DEBUG("m_jFexSumEtContainerKey"<< m_jFexSumEtContainerKey);

  ATH_CHECK( m_jFexLRJetContainerKey.initialize() );  
  ATH_CHECK( m_jFexSRJetContainerKey.initialize() );  
  ATH_CHECK( m_jFexTauContainerKey.initialize()   );  
  ATH_CHECK( m_jFexFwdElContainerKey.initialize()   );  
  ATH_CHECK( m_jFexMETContainerKey.initialize()   );  
  ATH_CHECK( m_jFexSumEtContainerKey.initialize()   );  

  return AthMonitorAlgorithm::initialize();
}

StatusCode JfexMonitorAlgorithm::fillHistograms( const EventContext& ctx ) const {
  
  ATH_MSG_DEBUG("JfexMonitorAlgorithm::fillHistograms");
  
  SG::ReadHandle<xAOD::jFexSRJetRoIContainer> jFexSRJetContainer{m_jFexSRJetContainerKey, ctx};
  if(!jFexSRJetContainer.isValid()){
    ATH_MSG_ERROR("No jFex SR Jet container found in storegate  "<< m_jFexSRJetContainerKey);
    return StatusCode::SUCCESS;
  }
  SG::ReadHandle<xAOD::jFexLRJetRoIContainer> jFexLRJetContainer{m_jFexLRJetContainerKey, ctx};
  if(!jFexLRJetContainer.isValid()){
    ATH_MSG_ERROR("No jFex LR Jet container found in storegate  "<< m_jFexLRJetContainerKey);
    return StatusCode::SUCCESS;
  }
  SG::ReadHandle<xAOD::jFexTauRoIContainer> jFexTauContainer{m_jFexTauContainerKey, ctx};
  if(!jFexTauContainer.isValid()){
    ATH_MSG_ERROR("No jFex Tau container found in storegate  "<< m_jFexTauContainerKey);
    return StatusCode::SUCCESS;
  }
  SG::ReadHandle<xAOD::jFexFwdElRoIContainer> jFexEMContainer{m_jFexFwdElContainerKey, ctx};
  if(!jFexEMContainer.isValid()){
    ATH_MSG_ERROR("No jFex EM container found in storegate  "<< m_jFexFwdElContainerKey);
    return StatusCode::SUCCESS;
  }
  SG::ReadHandle<xAOD::jFexMETRoIContainer> jFexMETContainer{m_jFexMETContainerKey, ctx};
  if(!jFexMETContainer.isValid()){
    ATH_MSG_ERROR("No jFex MET container found in storegate  "<< m_jFexMETContainerKey);
    return StatusCode::SUCCESS;
  }
  SG::ReadHandle<xAOD::jFexSumETRoIContainer> jFexSumETContainer{m_jFexSumEtContainerKey, ctx};
  if(!jFexSumETContainer.isValid()){
    ATH_MSG_ERROR("No jFex MET container found in storegate  "<< m_jFexSumEtContainerKey);
    return StatusCode::SUCCESS;
  }

  // variables for histograms
  // jJ
  auto jFexSRJetModule  = Monitored::Scalar<int>  ("jJ_jFexNumber",0);
  auto jFexSRJetFPGA    = Monitored::Scalar<int>  ("jJ_fpgaNumber",0);
  auto jFexSRJetEt  = Monitored::Scalar<int>  ("jJ_Et",0);
  auto jFexSRJeteta = Monitored::Scalar<float>("jJ_Eta",0.0);
  auto jFexSRJetphi = Monitored::Scalar<float>("jJ_Phi",0.0);
  auto jFexSRJeteta_glo = Monitored::Scalar<float>("jJ_GlobalEta",0.0);
  auto jFexSRJetphi_glo = Monitored::Scalar<float>("jJ_GlobalPhi",0.0);
  
  // jLJ
  auto jFexLRJetModule  = Monitored::Scalar<int>  ("jLJ_jFexNumber",0);
  auto jFexLRJetFPGA    = Monitored::Scalar<int>  ("jLJ_fpgaNumber",0);
  auto jFexLRJetEt  = Monitored::Scalar<int>  ("jLJ_Et",0);
  auto jFexLRJeteta = Monitored::Scalar<float>("jLJ_Eta",0.0);
  auto jFexLRJetphi = Monitored::Scalar<float>("jLJ_Phi",0.0);
  auto jFexLRJeteta_glo = Monitored::Scalar<float>("jLJ_GlobalEta",0.0);
  auto jFexLRJetphi_glo = Monitored::Scalar<float>("jLJ_GlobalPhi",0.0);
  
  
  // jTau
  auto jFexTauModule  = Monitored::Scalar<int>  ("jTau_jFexNumber",0);
  auto jFexTauFPGA    = Monitored::Scalar<int>  ("jTau_fpgaNumber",0);
  auto jFexTauEt  = Monitored::Scalar<int>  ("jTau_Et",0);
  auto jFexTauIso = Monitored::Scalar<int>  ("jTau_Iso",0);
  auto jFexTaueta = Monitored::Scalar<float>("jTau_Eta",0.0);
  auto jFexTauphi = Monitored::Scalar<float>("jTau_Phi",0.0);
  auto jFexTaueta_glo = Monitored::Scalar<float>("jTau_GlobalEta",0.0);
  auto jFexTauphi_glo = Monitored::Scalar<float>("jTau_GlobalPhi",0.0);
  
  // jEM
  auto jFexEMModule  = Monitored::Scalar<int>  ("jEM_jFexNumber",0);
  auto jFexEMFPGA    = Monitored::Scalar<int>  ("jEM_fpgaNumber",0);
  auto jFexEMEt  = Monitored::Scalar<int>  ("jEM_Et",0);
  auto jFexEMeta = Monitored::Scalar<float>("jEM_Eta",0.0);
  auto jFexEMphi = Monitored::Scalar<float>("jEM_Phi",0.0);
  auto jFexEMeta_glo = Monitored::Scalar<float>("jEM_GlobalEta",0.0);
  auto jFexEMphi_glo = Monitored::Scalar<float>("jEM_GlobalPhi",0.0);
  auto jFexEMIso = Monitored::Scalar<int>  ("jEM_Iso",0);
  auto jFexEMf1  = Monitored::Scalar<int>  ("jEM_f1",0);
  auto jFexEMf2  = Monitored::Scalar<int>  ("jEM_f2",0);
  
  // jXE
  auto jFexMETX   = Monitored::Scalar<int>  ("jXE_X",0);
  auto jFexMETY   = Monitored::Scalar<int>  ("jXE_Y",0);
  auto jFexMET    = Monitored::Scalar<float>("jXE_MET",0.0);
  auto jFexMETphi = Monitored::Scalar<float>("jXE_phi",0.0);
  
  // jTE
  auto jFexSumEt_low   = Monitored::Scalar<int>  ("jTE_low",0);
  auto jFexSumEt_high  = Monitored::Scalar<int>  ("jTE_high",0);
  auto jFexSumEt_total = Monitored::Scalar<float>("jTE_SumEt",0.0);

  for(const xAOD::jFexSRJetRoI* jFexSRJetRoI : *jFexSRJetContainer){
    if(jFexSRJetRoI->tobWord()==0) continue; //remove empty TOBs
    jFexSRJetModule=jFexSRJetRoI->jFexNumber();
    jFexSRJetFPGA=jFexSRJetRoI->fpgaNumber();
    jFexSRJetEt=jFexSRJetRoI->tobEt();
    jFexSRJeteta=jFexSRJetRoI->eta();
    jFexSRJetphi=jFexSRJetRoI->phi();
    jFexSRJeteta_glo=jFexSRJetRoI->globalEta();
    jFexSRJetphi_glo=jFexSRJetRoI->globalPhi();
    fill(m_packageName,jFexSRJetModule,jFexSRJetFPGA,jFexSRJetEt,jFexSRJeteta,jFexSRJetphi,jFexSRJeteta_glo,jFexSRJetphi_glo);
  }
  
  for(const xAOD::jFexLRJetRoI* jFexLRJetRoI : *jFexLRJetContainer){
    if(jFexLRJetRoI->tobWord()==0) continue; //remove empty TOBs
    jFexLRJetModule=jFexLRJetRoI->jFexNumber();
    jFexLRJetFPGA=jFexLRJetRoI->fpgaNumber();
    jFexLRJetEt=jFexLRJetRoI->tobEt();
    jFexLRJeteta=jFexLRJetRoI->eta();
    jFexLRJetphi=jFexLRJetRoI->phi();
    jFexLRJeteta_glo=jFexLRJetRoI->globalEta();
    jFexLRJetphi_glo=jFexLRJetRoI->globalPhi();
    fill(m_packageName,jFexLRJetModule,jFexLRJetFPGA,jFexLRJetEt,jFexLRJeteta,jFexLRJetphi,jFexLRJeteta_glo,jFexLRJetphi_glo);
  }
  
  for(const xAOD::jFexTauRoI* jFexTauRoI : *jFexTauContainer){
    if(jFexTauRoI->tobWord()==0) continue; //remove empty TOBs
    jFexTauModule =jFexTauRoI->jFexNumber();
    jFexTauFPGA =jFexTauRoI->fpgaNumber();
    jFexTauEt =jFexTauRoI->tobEt();
    jFexTauIso=jFexTauRoI->tobIso();
    jFexTaueta=jFexTauRoI->eta();
    jFexTauphi=jFexTauRoI->phi();
    jFexTaueta_glo=jFexTauRoI->globalEta();
    jFexTauphi_glo=jFexTauRoI->globalPhi();
    fill(m_packageName,jFexTauModule,jFexTauFPGA,jFexTauEt,jFexTauIso,jFexTaueta,jFexTauphi,jFexTaueta_glo,jFexTauphi_glo);
  }

  for(const xAOD::jFexFwdElRoI* jFexFwdElRoI : *jFexEMContainer){
    if(jFexFwdElRoI->tobWord()==0) continue; //remove empty TOBs
    jFexEMModule =jFexFwdElRoI->jFexNumber();
    jFexEMFPGA =jFexFwdElRoI->fpgaNumber();
    jFexEMEt =jFexFwdElRoI->tobEt();
    jFexEMeta=jFexFwdElRoI->eta();
    jFexEMphi=jFexFwdElRoI->phi();
    jFexEMeta_glo=jFexFwdElRoI->globalEta();
    jFexEMphi_glo=jFexFwdElRoI->globalPhi();
    jFexEMIso=jFexFwdElRoI->tobEMIso();
    jFexEMf1 =jFexFwdElRoI->tobEMf1();
    jFexEMf2 =jFexFwdElRoI->tobEMf2();
    fill(m_packageName,jFexEMModule,jFexEMFPGA,jFexEMEt,jFexEMeta,jFexEMphi,jFexEMeta_glo,jFexEMphi_glo,jFexEMIso,jFexEMf1,jFexEMf2);
  }

  int metx = 0;
  int mety = 0;
  for(const xAOD::jFexMETRoI* jFexMETRoI : *jFexMETContainer){
    jFexMETX =jFexMETRoI->tobEx();
    jFexMETY =jFexMETRoI->tobEy();
    metx += jFexMETRoI->tobEx();
    mety += jFexMETRoI->tobEy();
    fill(m_packageName,jFexMETX,jFexMETY);
  }
  if(jFexMETContainer->size()>0){
      jFexMET    = TMath::Sqrt(std::pow(metx,2)+std::pow(mety,2));
      jFexMETphi = TMath::ATan2(mety,metx);
      fill(m_packageName,jFexMET,jFexMETphi);      
  }

  
  int sumEt_total = 0;
  for(const xAOD::jFexSumETRoI* jFexSumETRoI : *jFexSumETContainer){
    jFexSumEt_low =jFexSumETRoI->tobEt_lower();
    jFexSumEt_high =jFexSumETRoI->tobEt_upper();
    sumEt_total += jFexSumETRoI->tobEt_lower()+jFexSumETRoI->tobEt_upper();
    fill(m_packageName,jFexSumEt_low,jFexSumEt_high);
  }
  if(jFexSumETContainer->size()>0){
      jFexSumEt_total = sumEt_total;
      fill(m_packageName,jFexSumEt_total);      
  }

    
  return StatusCode::SUCCESS;
}

File: Trigger/TrigT1/TrigT1CaloMonitoring/python/JfexMonitorAlgorithm.py
#
#  Copyright (C) 2002-2022 CERN for the benefit of the ATLAS collaboration
#
def JfexMonitoringConfig(inputFlags):
    '''Function to configure LVL1 Jfex algorithm in the monitoring system.'''

    import math 
    # get the component factory - used for getting the algorithms
    from AthenaConfiguration.ComponentFactory import CompFactory
    from AthenaConfiguration.ComponentAccumulator import ComponentAccumulator
    result = ComponentAccumulator()

    # make the athena monitoring helper
    from AthenaMonitoring import AthMonitorCfgHelper
    helper = AthMonitorCfgHelper(inputFlags,'JfexMonitoringCfg')

    # get any algorithms
    JfexMonAlg = helper.addAlgorithm(CompFactory.JfexMonitorAlgorithm,'JfexMonAlg')

    # add any steering
    groupName = 'JfexMonitor' # the monitoring group name is also used for the package name
    JfexMonAlg.PackageName = groupName

    mainDir = 'L1Calo'
    trigPath = 'Jfex/'

    # add monitoring algorithm to group, with group name and main directory 
    myGroup = helper.addGroup(JfexMonAlg, groupName , mainDir)

    # define gfex histograms
    
    ######  jJ  ######  
    myGroup.defineHistogram('jJ_jFexNumber;h_jJ_jFexNumber', title='jFex SRJet Module;Module number;Counts',
                            type='TH1I', path=trigPath+'jJ/', xbins=6,xmin=0,xmax=6)

    myGroup.defineHistogram('jJ_fpgaNumber;h_jJ_fpgaNumber', title='jFex SRJet FPGA;FPGA number;Counts',
                            type='TH1F', path=trigPath+'jJ/',xbins=4,xmin=0,xmax=4)
                            
    myGroup.defineHistogram('jJ_Et;h_jJ_Et', title='jFex SRJet Transverse Energy;tobEt [200 MeV Scale];Counts',
                            type='TH1I', path=trigPath+'jJ/', xbins=512,xmin=0,xmax=2048)

    myGroup.defineHistogram('jJ_Eta;h_jJ_Eta', title='jFex SRJet #eta;#eta;Counts',
                            type='TH1F', path=trigPath+'jJ/',xbins=100,xmin=-5.0,xmax=5.0)

    myGroup.defineHistogram('jJ_Phi;h_jJ_Phi', title='jFex SRJet #phi;#phi;Counts',
                            type='TH1F', path=trigPath+'jJ/',xbins=66,xmin=-math.pi,xmax=math.pi)
                            
    myGroup.defineHistogram('jJ_Eta,jJ_Phi;h_jJ_EtaPhiMap', title="jFex SRJet #eta vs #phi;#eta;#phi",
                            type='TH2F',path=trigPath+'jJ/', xbins=1000,xmin=-5.0,xmax=5.0,ybins=1000,ymin=-math.pi,ymax=math.pi)                            

    myGroup.defineHistogram('jJ_GlobalEta;h_jJ_GlobalEta', title='jFex SRJet Global #eta;#eta;Counts',
                            type='TH1F', path=trigPath+'jJ/',xbins=100,xmin=-50,xmax=50)

    myGroup.defineHistogram('jJ_GlobalPhi;h_jJ_GlobalPhi', title='jFex SRJet Global #phi;#phi;Counts',
                            type='TH1F', path=trigPath+'jJ/',xbins=67,xmin=-1,xmax=65)
                            
    myGroup.defineHistogram('jJ_GlobalEta,jJ_GlobalPhi;h_jJ_GlobalEtaPhiMap', title="jFex SRJet Global #eta vs #phi;(int) #eta;(int) #phi",
                            type='TH2F',path=trigPath+'jJ/', xbins=1000,xmin=-50,xmax=50,ybins=1000,ymin=-1,ymax=65)                            
                            
    ######  jLJ  ######  
    myGroup.defineHistogram('jLJ_jFexNumber;h_jLJ_jFexNumber', title='jFex LRJet Module;Module number;Counts',
                            type='TH1I', path=trigPath+'jLJ/', xbins=6,xmin=0,xmax=6)

    myGroup.defineHistogram('jLJ_fpgaNumber;h_jLJ_fpgaNumber', title='jFex LRJet FPGA;FPGA number;Counts',
                            type='TH1F', path=trigPath+'jLJ/',xbins=4,xmin=0,xmax=4)
                            
    myGroup.defineHistogram('jLJ_Et;h_jLJ_Et', title='jFex LRJet Transverse Energy;tobEt [200 MeV Scale];Counts',
                            type='TH1I', path=trigPath+'jLJ/',xbins=512,xmin=0,xmax=2048)

    myGroup.defineHistogram('jLJ_Eta;h_jLJ_Eta', title='jFex LRJet #eta;#eta;Counts',
                            type='TH1F', path=trigPath+'jLJ/',xbins=60,xmin=-3.0,xmax=3.0)

    myGroup.defineHistogram('jLJ_Phi;h_jLJ_Phi', title='jFex LRJet #phi;#phi;Counts',
                            type='TH1F', path=trigPath+'jLJ/',xbins=66,xmin=-math.pi,xmax=math.pi)
                            
    myGroup.defineHistogram('jLJ_Eta,jLJ_Phi;h_jLJ_EtaPhiMap', title="jFEX LRJet #eta vs #phi;#eta;#phi",
                            type='TH2F',path=trigPath+'jLJ/', xbins=1000,xmin=-5.0,xmax=5.0,ybins=1000,ymin=-math.pi,ymax=math.pi)                             

    myGroup.defineHistogram('jLJ_GlobalEta;h_jLJ_GlobalEta', title='jFex LRJet Global #eta;#eta;Counts',
                            type='TH1F', path=trigPath+'jLJ/',xbins=100,xmin=-50,xmax=50)

    myGroup.defineHistogram('jLJ_GlobalPhi;h_jLJ_GlobalPhi', title='jFex LRJet Global #phi;#phi;Counts',
                            type='TH1F', path=trigPath+'jLJ/',xbins=67,xmin=-1,xmax=65)
                            
    myGroup.defineHistogram('jLJ_GlobalEta,jLJ_GlobalPhi;h_jLJ_GlobalEtaPhiMap', title="jFex LRJet #eta vs #phi;(int) #eta; (int) #phi",
                            type='TH2F',path=trigPath+'jLJ/', xbins=1000,xmin=-50,xmax=50,ybins=1000,ymin=-1,ymax=65)                                  
    ######  jTau  ######   
    myGroup.defineHistogram('jTau_jFexNumber;h_jTau_jFexNumber', title='jFex Tau Module;Module number;Counts',
                            type='TH1I', path=trigPath+'jTau/', xbins=6,xmin=0,xmax=6)

    myGroup.defineHistogram('jTau_fpgaNumber;h_jTau_fpgaNumber', title='jFex Tau FPGA;FPGA number;Counts',
                            type='TH1F', path=trigPath+'jTau/',xbins=4,xmin=0,xmax=4)
                            
    myGroup.defineHistogram('jTau_Et;h_jTau_Et', title='jFex Tau Transverse Energy;tobEt [200 MeV Scale];Counts',
                            type='TH1I', path=trigPath+'jTau/',xbins=512,xmin=0,xmax=2048)
    
    myGroup.defineHistogram('jTau_Iso;h_jTau_Iso', title='jFex Tau Isolation;tobIso [200 MeV Scale];Counts',
                            type='TH1I', path=trigPath+'jTau/',xbins=512,xmin=0,xmax=2048)

    myGroup.defineHistogram('jTau_Eta;h_jTau_Eta', title='jFex Tau #eta;#eta;Counts',
                            type='TH1F', path=trigPath+'jTau/',xbins=60,xmin=-3.0,xmax=3.0)

    myGroup.defineHistogram('jTau_Phi;h_jTau_Phi', title='jFex Tau #phi;#phi;Counts',
                            type='TH1F', path=trigPath+'jTau/',xbins=66,xmin=-math.pi,xmax=math.pi)
                            
    myGroup.defineHistogram('jTau_Eta,jTau_Phi;h_jTau_EtaPhiMap', title="jFex Tau #eta vs #phi;#eta;#phi",
                            type='TH2F',path=trigPath+'jTau/', xbins=1000,xmin=-5.0,xmax=5.0,ybins=1000,ymin=-math.pi,ymax=math.pi)          

    myGroup.defineHistogram('jTau_GlobalEta;h_jTau_GlobalEta', title='jFex Tau Global #eta;#eta;Counts',
                            type='TH1F', path=trigPath+'jTau/',xbins=100,xmin=-50,xmax=50)

    myGroup.defineHistogram('jTau_GlobalPhi;h_jTau_GlobalPhi', title='jFex Tau Global #phi;#phi;Counts',
                            type='TH1F', path=trigPath+'jTau/',xbins=67,xmin=-1,xmax=65)
                            
    myGroup.defineHistogram('jTau_GlobalEta,jTau_GlobalPhi;h_jTau_GlobalEtaPhiMap', title="jFex Tau Global #eta vs #phi;(int) #eta; (int) #phi",
                            type='TH2F',path=trigPath+'jTau/', xbins=1000,xmin=-50,xmax=50,ybins=1000,ymin=-1,ymax=65)                               
    ######  jEM  ######  
    myGroup.defineHistogram('jEM_jFexNumber;h_jEM_jFexNumber', title='jFex EM Module;Module number;Counts',
                            type='TH1I', path=trigPath+'jEM/', xbins=6,xmin=0,xmax=6)

    myGroup.defineHistogram('jEM_fpgaNumber;h_jEM_fpgaNumber', title='jFex EM FPGA;FPGA number;Counts',
                            type='TH1F', path=trigPath+'jEM/',xbins=4,xmin=0,xmax=4)
                            
    myGroup.defineHistogram('jEM_Et;h_jEM_Et', title='jFex EM Transverse Energy;tobEt [200 MeV Scale];Counts',
                            type='TH1I', path=trigPath+'jEM/',xbins=512,xmin=0,xmax=2048)

    myGroup.defineHistogram('jEM_Eta;h_jEM_Eta', title='jFex EM #eta;#eta;Counts',
                            type='TH1F', path=trigPath+'jEM/',xbins=60,xmin=-3.0,xmax=3.0)

    myGroup.defineHistogram('jEM_Phi;h_jEM_Phi', title='jFex EM #phi;#phi;Counts',
                            type='TH1F', path=trigPath+'jEM/',xbins=66,xmin=-math.pi,xmax=math.pi)

    em_labels = ['None','loose','medium','tight']
    myGroup.defineHistogram('jEM_Iso;h_jEM_Iso', title='jFex EM Isolation;tobIso;Counts',
                            type='TH1I', path=trigPath+'jEM/',xbins=4,xmin=0,xmax=4,xlabels=em_labels)

    myGroup.defineHistogram('jEM_f1;h_jEM_f1', title='jFex EM Frac1;EM Frac1;Counts',
                            type='TH1I', path=trigPath+'jEM/',xbins=4,xmin=0,xmax=4,xlabels=em_labels)

    myGroup.defineHistogram('jEM_f2;h_jEM_f2', title='jFex EM Frac2;EM Frac2;Counts',
                            type='TH1I', path=trigPath+'jEM/',xbins=4,xmin=0,xmax=4,xlabels=em_labels)
                            
    myGroup.defineHistogram('jEM_Eta,jEM_Phi;h_jEM_EtaPhiMap', title="jFex EM #eta vs #phi;#eta;#phi",
                            type='TH2F',path=trigPath+'jEM/', xbins=1000,xmin=-5.0,xmax=5.0,ybins=1000,ymin=-math.pi,ymax=math.pi) 

    myGroup.defineHistogram('jEM_GlobalEta;h_jEM_GlobalEta', title='jFex EM Global #eta;#eta;Counts',
                            type='TH1F', path=trigPath+'jEM/',xbins=100,xmin=-50,xmax=50)

    myGroup.defineHistogram('jEM_GlobalPhi;h_jEM_GlobalPhi', title='jFex EM Global #phi;#phi;Counts',
                            type='TH1F', path=trigPath+'jEM/',xbins=67,xmin=-1,xmax=65)
                            
    myGroup.defineHistogram('jEM_GlobalEta,jEM_GlobalPhi;h_jEM_GlobalEtaPhiMap', title="jFex EM Global #eta vs #phi;(int) #eta; (int) #phi",
                            type='TH2F',path=trigPath+'jEM/', xbins=1000,xmin=-50,xmax=50,ybins=1000,ymin=-1,ymax=65)                                
    ######  jXE  ######                          
    myGroup.defineHistogram('jXE_X;h_jXE_X', title='jFex MET X component;tobEx [200 MeV Scale];Counts',
                            type='TH1I', path=trigPath+'jXE/',xbins=1000,xmin=-500,xmax=500)
    
    myGroup.defineHistogram('jXE_Y;h_jXE_Y', title='jFex MET Y component;tobEy [200 MeV Scale];Counts',
                            type='TH1I', path=trigPath+'jXE/',xbins=1000,xmin=-500,xmax=500)

    myGroup.defineHistogram('jXE_MET;h_jXE_MET', title='jFex MET ;Total jXE [200 GeV Scale];Counts',
                            type='TH1F', path=trigPath+'jXE/',xbins=500,xmin=0,xmax=1000)

    myGroup.defineHistogram('jXE_phi;h_jXE_phi', title='jFex MET phi ;#phi=atan(Ey/Ex) values;Counts',
                            type='TH1F', path=trigPath+'jXE/',xbins=66,xmin=-math.pi,xmax=math.pi)

                            
    ######  jTE  ######                          
    myGroup.defineHistogram('jTE_low;h_jTE_low', title='jFex SumEt low #eta;tob Et_lower [GeV/200];Counts',
                            type='TH1I', path=trigPath+'jTE/',xbins=700,xmin=0,xmax=700)
    
    myGroup.defineHistogram('jTE_high;h_jTE_high', title='jFex SumEt high #eta;tob Et_upper [GeV/200];Counts',
                            type='TH1I', path=trigPath+'jTE/',xbins=700,xmin=0,xmax=700)

    myGroup.defineHistogram('jTE_SumEt;h_jTE_SumEt', title='jFex SumEt total ;Total jTE [200 GeV Scale];Counts',
                            type='TH1F', path=trigPath+'jTE/',xbins=1000,xmin=0,xmax=4000)


    acc = helper.result()
    result.merge(acc)
    return result


if __name__=='__main__':
    # set input file and config options
    from AthenaConfiguration.AllConfigFlags import ConfigFlags
    import glob
    inputs = glob.glob('/eos/user/t/thompson/ATLAS/LVL1_mon/MC_ESD/mc21.601189.PhPy8EG_AZNLO_Zee.recon.ESD.e8392_e7400_s3775_r13614_r13689/*root*')

    ConfigFlags.Input.Files = inputs
    ConfigFlags.Output.HISTFileName = 'ExampleMonitorOutput_LVL1.root'

    ConfigFlags.Exec.MaxEvents=10

    ConfigFlags.lock()
    ConfigFlags.dump() # print all the configs

    from AthenaCommon.AppMgr import ServiceMgr
    ServiceMgr.Dump = False

    from AthenaConfiguration.MainServicesConfig import MainServicesCfg 
    from AthenaPoolCnvSvc.PoolReadConfig import PoolReadCfg
    cfg = MainServicesCfg(ConfigFlags)
    cfg.merge(PoolReadCfg(ConfigFlags))

    JfexMonitorCfg = JfexMonitoringConfig(ConfigFlags)
    cfg.merge(JfexMonitorCfg)
    cfg.run()
